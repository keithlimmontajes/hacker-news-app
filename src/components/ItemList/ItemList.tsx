import React from 'react';
import Card from '../Card';
import moment from 'moment';
import {TouchableOpacity} from 'react-native';
import {Text, View, XStack} from 'tamagui';
import {ItemType} from '../../screens/Home/types';
import {useGetNewsAuthorQuery} from '../../services/news';
import {useNavigation} from '@react-navigation/native';

const ItemList = ({item}: ItemType) => {
  const {data: author} = useGetNewsAuthorQuery(item?.by);
  const navigation = useNavigation();

  return (
    <View>
      <Card>
        <Text fontSize={18} py={5} fontWeight={'500'}>
          {item?.title}
        </Text>

        <TouchableOpacity
          // @ts-ignore
          onPress={() => navigation.navigate('Details', {url: item?.url})}>
          <Text color={'#FB651E'}>{item?.url}</Text>
        </TouchableOpacity>

        <Text pt={5} fontSize={12}>
          Author:{' '}
          <Text fontSize={14} color="gray">
            {item?.by}
          </Text>
        </Text>

        <XStack gap={'$1'} pt="$4">
          <Text color="gray">
            ⧗ {moment.unix(item.time).format('H') + 'h'} |
          </Text>
          <Text color="gray">✰ scores: {item?.score} |</Text>
          <Text color="gray">♡ {author?.karma}</Text>
        </XStack>
      </Card>
    </View>
  );
};

export default React.memo(ItemList);
