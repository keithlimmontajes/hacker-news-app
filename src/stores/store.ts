import {configureStore} from '@reduxjs/toolkit';
import {persistStore, persistReducer} from 'redux-persist';
import {apiMiddleware} from './middleware';
import rootReducer from './rootReducer';
import AsyncStorage from '@react-native-async-storage/async-storage';

const persistConfig = {
  key: 'root',
  storage: AsyncStorage,
  whitelist: ['newsSlice'],
};

const persistedReducer = persistReducer(persistConfig, rootReducer);

const store = configureStore({
  reducer: persistedReducer,
  middleware: getDefaultMiddleware =>
    getDefaultMiddleware({
      serializableCheck: {
        ignoredActions: ['persist/PERSIST', 'persist/REHYDRATE'],
      },
    }).concat(apiMiddleware),
});

export const persistor = persistStore(store);

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;

export default store;
