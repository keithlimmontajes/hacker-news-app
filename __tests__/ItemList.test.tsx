import React from 'react';
import {render} from '@testing-library/react-native';
import ItemList from '../src/components/ItemList';
import {Provider} from 'react-redux';
import store from '../src/stores/store';
import {NavigationContainer} from '@react-navigation/native';
import {TamaguiProvider} from 'tamagui';
import config from '../src/themes';
import {ThemeProvider} from '@tamagui/core';

describe('ItemListComponent', () => {
  it('Should work as expected', () => {
    const all = render(
      <Provider store={store}>
        <ThemeProvider defaultTheme="light">
          <NavigationContainer>
            <TamaguiProvider config={config}>
              <ItemList
                item={{
                  id: '',
                  title: '',
                  url: '',
                  time: undefined,
                  score: '',
                  by: '',
                }}
              />
            </TamaguiProvider>
          </NavigationContainer>
        </ThemeProvider>
      </Provider>,
    );
    expect(all.toJSON()).toMatchSnapshot();
  });
});
