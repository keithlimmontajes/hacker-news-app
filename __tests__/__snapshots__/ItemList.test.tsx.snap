// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardComponent Should work as expected 1`] = `
<View>
  <View
    style={
      {
        "backgroundColor": "white",
        "borderBottomColor": "#E9E9E9",
        "borderBottomLeftRadius": 8,
        "borderBottomRightRadius": 8,
        "borderBottomWidth": 1,
        "borderLeftColor": "#E9E9E9",
        "borderLeftWidth": 1,
        "borderRightColor": "#E9E9E9",
        "borderRightWidth": 1,
        "borderStyle": "solid",
        "borderTopColor": "#E9E9E9",
        "borderTopLeftRadius": 8,
        "borderTopRightRadius": 8,
        "borderTopWidth": 1,
        "marginBottom": 7,
        "marginTop": 7,
        "paddingBottom": 7,
        "paddingLeft": 7,
        "paddingRight": 7,
        "paddingTop": 7,
      }
    }
  >
    <Text
      style={
        {
          "color": "hsla(0, 0%, 9%, 1)",
          "fontFamily": "System",
          "fontSize": 18,
          "fontWeight": "500",
          "paddingBottom": 5,
          "paddingTop": 5,
        }
      }
      suppressHighlighting={true}
    />
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          {
            "color": "#FB651E",
            "fontFamily": "System",
          }
        }
        suppressHighlighting={true}
      />
    </View>
    <Text
      style={
        {
          "color": "hsla(0, 0%, 9%, 1)",
          "fontFamily": "System",
          "fontSize": 12,
          "paddingTop": 5,
        }
      }
      suppressHighlighting={true}
    >
      Author:
       
      <Text
        style={
          {
            "color": "gray",
            "fontFamily": "System",
            "fontSize": 14,
          }
        }
        suppressHighlighting={true}
      />
    </Text>
    <View
      style={
        {
          "flexDirection": "row",
          "gap": 2,
          "paddingTop": 18,
        }
      }
    >
      <Text
        style={
          {
            "color": "gray",
            "fontFamily": "System",
          }
        }
        suppressHighlighting={true}
      >
        ⧗ 
        Invalid dateh
         |
      </Text>
      <Text
        style={
          {
            "color": "gray",
            "fontFamily": "System",
          }
        }
        suppressHighlighting={true}
      >
        ✰ scores: 
         |
      </Text>
      <Text
        style={
          {
            "color": "gray",
            "fontFamily": "System",
          }
        }
        suppressHighlighting={true}
      >
        ♡ 
      </Text>
    </View>
  </View>
</View>
`;

exports[`ItemListComponent Should work as expected 1`] = `
<View>
  <View
    style={
      {
        "backgroundColor": "white",
        "borderBottomColor": "#E9E9E9",
        "borderBottomLeftRadius": 8,
        "borderBottomRightRadius": 8,
        "borderBottomWidth": 1,
        "borderLeftColor": "#E9E9E9",
        "borderLeftWidth": 1,
        "borderRightColor": "#E9E9E9",
        "borderRightWidth": 1,
        "borderStyle": "solid",
        "borderTopColor": "#E9E9E9",
        "borderTopLeftRadius": 8,
        "borderTopRightRadius": 8,
        "borderTopWidth": 1,
        "marginBottom": 7,
        "marginTop": 7,
        "paddingBottom": 7,
        "paddingLeft": 7,
        "paddingRight": 7,
        "paddingTop": 7,
      }
    }
  >
    <Text
      style={
        {
          "color": "hsla(0, 0%, 9%, 1)",
          "fontFamily": "System",
          "fontSize": 18,
          "fontWeight": "500",
          "paddingBottom": 5,
          "paddingTop": 5,
        }
      }
      suppressHighlighting={true}
    />
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          {
            "color": "#FB651E",
            "fontFamily": "System",
          }
        }
        suppressHighlighting={true}
      />
    </View>
    <Text
      style={
        {
          "color": "hsla(0, 0%, 9%, 1)",
          "fontFamily": "System",
          "fontSize": 12,
          "paddingTop": 5,
        }
      }
      suppressHighlighting={true}
    >
      Author:
       
      <Text
        style={
          {
            "color": "gray",
            "fontFamily": "System",
            "fontSize": 14,
          }
        }
        suppressHighlighting={true}
      />
    </Text>
    <View
      style={
        {
          "flexDirection": "row",
          "gap": 2,
          "paddingTop": 18,
        }
      }
    >
      <Text
        style={
          {
            "color": "gray",
            "fontFamily": "System",
          }
        }
        suppressHighlighting={true}
      >
        ⧗ 
        Invalid dateh
         |
      </Text>
      <Text
        style={
          {
            "color": "gray",
            "fontFamily": "System",
          }
        }
        suppressHighlighting={true}
      >
        ✰ scores: 
         |
      </Text>
      <Text
        style={
          {
            "color": "gray",
            "fontFamily": "System",
          }
        }
        suppressHighlighting={true}
      >
        ♡ 
      </Text>
    </View>
  </View>
</View>
`;
